portal-web/test/functional/com/liferay/portalweb/macros/LiferayEvent.macro

```diff
@@ -75,4 +75,31 @@ window.LFR_JS_EVENTS_LOG = {
 		RunScript(locator1 = "${javascript}");
 	}
 
+	macro waitForLiferayEvent {
+		var javascript = '''
+try {
+	var eventValue = LFR_JS_EVENTS_LOG.getLastEvent("${eventName}").${attributeName};
+
+	if (eventValue.includes("${attributeValue}")) {
```

Comment:
You're assuming `eventValue` is a String with the `.includes` value. That's likely not always the case. This will throw `[eventValue].includes is not a function` when the value is not a String.

Reply:
I was just matching the logic here (which appears to expect a string)
https://github.com/wincent/liferay-portal/pull/248/commits/8484abb6fbd0e2b0f113b26844589bbd1fcab46e#diff-31caf4f08d87e0d6057c6efad73f9b8dR3810

If the value isn't a string, you'd also end up with a ClassCastException in Java. 

The JS you guys can rewrite to be shaped according to what makes sense though.